library/connect.c

```diff
@@ -191,10 +194,31 @@ static int ziti_connect(struct ziti_ctx *ctx, const ziti_net_session *session, s
     conn->token = session->token;
 
     ziti_edge_router **er;
+    ziti_channel_t *best_ch = NULL;
+    uint64_t best_latency = UINT64_MAX;
+
+    char ch_name[128];
     for (er = session->edge_routers; *er != NULL; er++) {
+        snprintf(ch_name, sizeof(ch_name), "%s@%s", (*er)->name, (*er)->ingress.tls);
+        ziti_channel_t *ch = model_map_get(&ctx->channels, ch_name);
+
+        if (ch != NULL && ch->state == Connected) {
+            if (ch->latency < best_latency) {
+                best_ch = ch;
+                best_latency = ch->latency;
+            }
+        }
+        else {
+            req->chan_tries++;
+            ZITI_LOG(TRACE, "connecting to %s(%s) for session[%s]", (*er)->name, (*er)->ingress.tls, conn->token);
+            ziti_channel_connect(ctx, (*er)->name, (*er)->ingress.tls, on_channel_connected, req);
+        }
+    }
+
+    if (best_ch) {
+        ZITI_LOG(DEBUG, "selected ch[%s] for best latency(%ldms)", best_ch->ingress, best_ch->latency);
         req->chan_tries++;
-        ZITI_LOG(TRACE, "connecting to %s(%s) for session[%s]", (*er)->name, (*er)->ingress.tls, conn->token);
-        ziti_channel_connect(ctx, (*er)->ingress.tls, on_channel_connected, req);
+        on_channel_connected(best_ch, req, ZITI_OK);
     }
```

Comment:
seems like a warn if there's no best channel selected? why keep the req->chan_tries line?